### PR TITLE
feat: run app using docker

### LINKED ISSUE #8591: Create a full independent docker environment to run the project on
### Is your proposal related to a problem?

I tried to get one "good first issue" and was kind of annoying to setup a database externally just to start to contribute to the project

### Describe the solution you'd like

It would be nice to have a `docker-compose.yml` setup with everything that we need to run the app, something that with a `docker-compose up` on the repo's source we'll be able to run the project easily.

### PR DESCRIPTION
**Pull request by wesleymatosdev**
- feat: create files to copy the environment variables
- feat: create docker file
- feat: run docker-compose instead of yarn dev

## What does this PR do?

This PR aims to facilitate new contributions to the repo. We already have the `yarn dx` command to set up the database and the dependencies, but we still need to handle things like the environment variables manually.

After this PR, this will be done automatically following the instructions on the setup doc, and just by running the project will run with a few users registered.

Fixes #8591 (issue)

/claim #8591 

## Type of change

- Chore (refactoring code, technical debt, workflow improvements)
- New feature (non-breaking change which adds functionality)

## How should this be tested?

- [ ] Clone the repo, install the dependencies, and run `yarn dx`; the app should start correctly without any further action from the user (developer)

## Checklist

- ~I haven't checked if my PR needs changes to the documentation~ I checked, and it probably needs changes on the docs.

### CODE REVIEW

File: apps/web/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+COPY ./.env.appStore.example ./.env.appStore.example
+
+RUN cp ./.env.example ./.env
+RUN cp ./.env.appStore.example ./.env.appStore
+
+# Create NEXTAUTH_SECRET on .env
+RUN echo NEXTAUTH_SECRET=\"$(openssl rand -base64 32)\" >> ./.env
+
+# Create CALENDSO_ENCRYPTION_KEY on .env
+RUN echo CALENDSO_ENCRYPTION_KEY=\"$(openssl rand -base64 32)\" >> ./.env
```
Comment (Mohith234) [inline apps/web/Dockerfile:15]: Isn't this supposed to be ? `openssl rand -base64 24`

I saw the docs mentioning this command. Not sure if I'm correct.

Comment (ologbonowiwi) [inline apps/web/Dockerfile:15]:
```suggestion
RUN echo CALENDSO_ENCRYPTION_KEY=\"$(openssl rand -base64 24)\" >> ./.env
```

Comment (ologbonowiwi) [reply apps/web/Dockerfile]: Fixed; thank you for the feedback!

**Comment by emrysal** [comment]
Hi @ologbonowiwi please see the @calcom/docker project; that is where docker related configuration is stored and maintained 🙏

**Comment by ologbonowiwi** [comment]
@emrysal, I didn't intend to replace calcom/rocker with this implementation; it's my second PR trying to solve #8591, and it is frustrating to see my PRs being closed with a justification that doesn't make sense.

@PeerRich, should we close #8591? I'm encountering a lot of resistance to getting it done, and I don't know what else to do to solve this issue since you folks are saying to me that I should look to calcom/docker (which this PR uses, by the way), but calcom/docker doesn't help me to run the app without doing a bunch of stuff manually

**Comment by krumware** [comment]
@ologbonowiwi apologies for the resistance, there just has already been momentum on that side of the project and it will take some energy to change course. A conscious decision was made at one point to move the docker configuration out to another repo. If we focus on enhancing the documentation of the developer experience in that repo and calcom/docs, then we can make some progress there then make the decision to move things back into the repository here when the dust has settled. I'd appreciate your support over there, and also welcome it in the docker channel in the cal.com Slack.

(for the record, I share the opinion that the docker configuration is more appropriate in this repository, but there is a submodule pattern that was implemented by the core team that is also being respected here)

**Comment by ologbonowiwi** [comment]
@krumware I would love to be a part of the conversation and help in any way I can. I would greatly appreciate an invitation to the cal.com Slack channel, and my e-mail is ologbonowiwi520@gmail.com. Improving the developer experience is something that I am passionate about, and I'm excited to contribute to that. I appreciate your consideration and look forward to working with you all.

**Comment by krumware** [comment]
Sounds good! Here's the link to the calcom slack invite page https://cal.com/slack

**Comment by ologbonowiwi** [comment]
Unfortunately, I see the error `The email address must match one of the domains listed below. Please try another email.`

It looks like I'm unable to join because I don't have a cal.com/calendso e-mail; probably it'll be needed to send an invitation manually

**Comment by rjackson** [comment]
> but we still need to handle things like the environment variables manually.

I believe these are appropriately documented in the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md#developing), and whilst we might be able to automate the initial `.env` set up I don't think it's necessarily desirable to do so. Two main reasons:

1. In my opinion, its good to have developers set up their `.env` file themselves. This introduces them to the file and the fact that's where all their configuration lives.
2. If we have devs with intentionally different or customised configs, we don't want the automation undoing their intended changes.

If we did want to offer a shortcut to automate that initial set up, a smaller and easier-to-digest solution could be introducing a new `package.json` script. Something like:

```diff
     "dev:website": "turbo run dev --scope=\"@calcom/web\" --scope=\"@calcom/website\"",
     "dev": "turbo run dev --scope=\"@calcom/web\"",
     "build-storybook": "turbo run build --scope=\"@calcom/storybook\"",
+    "init-env": "[ ! -f .env ] && cp .env.example .env && sed -i '' \"/^NEXTAUTH_SECRET=\\$/ s|\\$|$(openssl rand -base64 32)|\" .env && sed -i '' \"/^CALENDSO_ENCRYPTION_KEY=\\$/ s|\\$|$(openssl rand -base64 24)|\" .env",
     "dx": "turbo run dx",
     "i-dev": "infisical run -- turbo run dev --scope=\"@calcom/web\"",
     "i-dx": "infisical run -- turbo run dx",
```

And updating README.md and CONTRIBUTING.md to suggest `yarn init-env` instead?

**Pull request closed without merging**